init.lua

```diff
@@ -119,12 +126,18 @@ end
 
 -- Register punchplayer callback.
 minetest.register_on_punchplayer(function(player, hitter, time_from_last_punch, tool_capabilities, dir, damage)
+
+	-- check if hitter != Player 
+	if hitter:is_player() == false then
+		return monster_do_no_damage 
+	end
+	
 	for k, v in pairs(pvp_areas_store:get_areas_for_pos(player:getpos())) do
 		if k then
-			return KILL_NO
+			return AREA_ACTIVATE 
```

Comment:
This should be `KILL_NO`. The function is `on_playerhit`, so the response should match:

`on_playerhit ` should thus return whether the player should receive damage, not whether an area gets activated...

Reply:
yes, but the KILL_NO or KILL_YES changes if safemode is aktivated

Reply:
for a better understanding of the code it would be better ro rename the variable
AREA_ACTIVATE to something like AREA_ACTIVATE_DAMAGE_RULE